src/Microsoft.Identity.Web.DownstreamRestApi/DownstreamRestApi.cs

```diff
@@ -326,44 +216,68 @@ internal class DownstreamRestApi : IDownstreamRestApi
 #endif
             }
 
-            string content = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+            HttpContent content = response.Content;
 
-            if (string.IsNullOrWhiteSpace(content))
+            if (content == null)
             {
                 return default;
             }
 
-            return JsonSerializer.Deserialize<TOutput>(content, new JsonSerializerOptions { PropertyNameCaseInsensitive = true });
+            if (effectiveOptions.Deserializer != null)
+            {
+                return effectiveOptions.Deserializer(content) as TOutput;
+            }
+            else
+            {
+                string stringContent = await content.ReadAsStringAsync();
+                return JsonSerializer.Deserialize<TOutput>(stringContent, new JsonSerializerOptions { PropertyNameCaseInsensitive = true });
+            }
         }
 
-        private static Action<DownstreamRestApiOptions> PrepareOptions(
-          Action<DownstreamRestApiOptions>? downstreamRestApiOptionsOverride,
-          HttpMethod httpMethod)
+        private async Task<HttpResponseMessage> CallRestApiInternalAsync(
+            string? serviceName,
+            DownstreamRestApiOptions effectiveOptions,
+            bool appToken,
+            HttpContent? content = null,
+            ClaimsPrincipal? user = null,
+            CancellationToken cancellationToken = default)
         {
-            Action<DownstreamRestApiOptions> downstreamRestApiOptions;
-
-            if (downstreamRestApiOptionsOverride == null)
+            if (effectiveOptions.Scopes == null)
```

Comment:
>effectiveOptions [](http://example.com/codeflow?start=16&length=16)

can these be null?

Reply:
not sure how we could know? Do you have an idea?